libs/basic_execution/tests/unit/basic_receiver.cpp

```diff
@@ -0,0 +1,288 @@
+        "mylib::receiver_3 should not be a receiver of a std::string");
+
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_1>::value,
+        "mylib::non_receiver_1 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_1,
+                      int>::value,
+        "mylib::non_receiver_1 should not be a receiver of int");
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_2>::value,
+        "mylib::non_receiver_2 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_2,
+                      int>::value,
+        "mylib::non_receiver_2 should not be a receiver of int");
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_3>::value,
+        "mylib::non_receiver_3 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_3,
+                      int>::value,
+        "mylib::non_receiver_3 should not be a receiver of int");
+    static_assert(
+        hpx::basic_execution::traits::is_receiver<mylib::non_receiver_4>::value,
+        "mylib::non_receiver_4 should be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_4,
+                      int>::value,
+        "mylib::non_receiver_4 should not be a receiver of int");
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_5>::value,
+        "mylib::non_receiver_5 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_5,
+                      int>::value,
+        "mylib::non_receiver_5 should not be a receiver of int");
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_6>::value,
+        "mylib::non_receiver_6 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_6,
+                      int>::value,
+        "mylib::non_receiver_6 should not be a receiver of int");
+    static_assert(
+        !hpx::basic_execution::traits::is_receiver<mylib::non_receiver_7>::value,
+        "mylib::non_receiver_7 should not be a receiver");
+    static_assert(!hpx::basic_execution::traits::is_receiver_of<mylib::non_receiver_7,
+                      int>::value,
+        "mylib::non_receiver_7 should not be a receiver of int");
+
+    {
+        mylib::receiver_1 rcv;
+        hpx::basic_execution::set_done(std::move(rcv));
+        HPX_TEST(done_called);
+        done_called = false;
+        hpx::basic_execution::set_error(std::move(rcv), std::exception_ptr{});
+        HPX_TEST(error_called);
+        error_called = false;
+        hpx::basic_execution::set_value(std::move(rcv), 1);
+        HPX_TEST(value_called);
+        value_called = false;
```

Comment:
`clang-tidy` doesn't like the multiple moves of `rcv`: https://app.circleci.com/pipelines/github/STEllAR-GROUP/hpx/3682/workflows/68cf2de8-a809-4344-b23c-0abc3c4355c5/jobs/194028/steps (with good reason, it looks like).